Source/JavaScriptCore/API/JSManagedValue.mm

```diff
@@ -149,11 +152,10 @@ - (JSValue *)value
         return nil;
 
     WTF::Locker<JSC::JSLock> locker(m_lock.get());
-    JSC::VM* vm = m_lock->vm();
```

Comment:
Isn't this idiom kinda wrong? If `vm` were the last reference to the VM at the end if this function then it would get derefed/destroyed before `locker` unlocks and we'd have a UAF. 

IMO, we should just use `JSC::JSLockHolder` and disallow using `WTF::Locker` with `JSLock` but that's a different patch.

Reply:
Yeah, this is fine in this case since VM's lifetime is independent from the API lock's.